R/user_utils.R

```diff
@@ -267,6 +267,247 @@ yn_to_numeric <- function(arg) {
   )
 }
 
+#' Convert XXTPT Strings to Hours
+#'
+#' Converts CDISC timepoint strings (e.g., `PCTPT`, `VSTPT`, `EGTPT`, `ISTPT`, `LBTPT`)
+#' into numeric hours for analysis. The function handles common dose-centric formats
+#' including pre-dose, post-dose (hours/minutes), days, time ranges, and special
+#' time markers.
+#'
+#' @param xxtpt A character vector of timepoint descriptions from SDTM `--TPT` variables
+#'   (e.g., `PCTPT`, `VSTPT`, `EGTPT`, `ISTPT`, `LBTPT`). Can contain `NA` values.
+#'
+#' @details
+#' The function recognizes the following patterns (all case-insensitive):
+#'
+#' **Special Cases:**
+#' * `"Screening"` → -1
+#' * `"Pre-dose"`, `"Predose"`, `"Pre-infusion"`, `"Infusion"`, `"0H"`, `"0 H"` → 0
+#' * `"EOI"`, `"End of Infusion"` → 1
+#' * `"Morning"`, `"Evening"` → `NA_real_`
+#' * Unrecognized values → `NA_real_`
+#'
+#' **Time-based Conversions:**
+#' * **Days**: `"Day 1"`, `"2D"`, `"2 days"`, `"1.5 days"` → multiply by 24 (e.g., 24, 48, 36)
+#' * **Hours + Minutes**: `"1H30M"`, `"1 hour 30 min"`, `"2HR15MIN"` → hours +
+#' minutes/60 (e.g., 1.5, 2.25)
+#' * **Time Ranges**: `"0-6h"`, `"6-12h Post-dose"`, `"0.5 - 6.5h"` → end value (e.g., 6, 12, 6.5)
+#' * **Hours only**: `"1H"`, `"2 hours"`, `"0.5HR"`, `"4hr Post-dose"` → as-is (e.g., 1, 2, 0.5, 4)
+#' * **Minutes only**: `"30M"`, `"45 min"`, `"2.5 Min"` → divide by 60 (e.g., 0.5, 0.75, 0.042)
+#'
+#' **Supported Unit Formats:**
+#' * Hours: H, h, HR, hr, HOUR, hour, HOURS, hours
+#' * Minutes: M, m, MIN, min, MINUTE, minute, MINUTES, minutes
+#' * Days: D, d, DAY, day, DAYS, days
+#' * All with optional plurals and flexible whitespace
+#' * Optional "Post-dose" or "Post dose" suffix supported for all patterns
+#'
+#' **Processing Order:**
+#' The function processes patterns in this order to avoid conflicts:
+#' 1. Special cases (exact matches)
+#' 2. Days
+#' 3. Hours + Minutes combinations
+#' 4. Time ranges (returns end value)
+#' 5. Hours only
+#' 6. Minutes only
```

Comment:
I think we don't need this section. All regexps are anchored. Thus there shouldn't be any conflicts.